cmd/bnsd/app/app_test.go

```diff
@@ -193,6 +195,65 @@ func sendToken(t *testing.T, baseApp abci.Application, chainID string, height in
 	return res
 }
 
+// checks batchWorks
+func sendBatch(t *testing.T, fail bool, baseApp weaveApp.BaseApp, chainID string, height int64, signers []Signer,
+	from, to []byte, amount int64, ticker, memo string, contracts ...[]byte) abci.ResponseDeliverTx {
+	msg := &cash.SendMsg{
+		Metadata: &weave.Metadata{Schema: 1},
+		Src:      from,
+		Dest:     to,
+		Amount: &coin.Coin{
+			Whole:  amount,
+			Ticker: ticker,
+		},
+		Memo: memo,
+	}
+
+	var messages []app.BatchMsg_Union
+	for i := 0; i < batch.MaxBatchMessages; i++ {
+		messages = append(messages,
+			app.BatchMsg_Union{
+				Sum: &app.BatchMsg_Union_SendMsg{
+					SendMsg: msg,
+				},
+			})
+	}
+
+	if fail == true {
+		messages[batch.MaxBatchMessages-1] = app.BatchMsg_Union{
```

Comment:
PP: I was confused to find an escrow msg here. Please add a comment why you do this